node-groups-enclave-tracks.tf

```
@@ -6,7 +6,7 @@ resource "aws_launch_template" "enclave_track" {
 
   name_prefix = "eks-node-enclaves-${each.key}-${var.cluster_name}-"
 
-  image_id                             = data.aws_ssm_parameter.al2023_ami[try(var.eks_node_group.arch, "amd64")].value
+  image_id                             = data.aws_ssm_parameter.al2023_ami[try(each.value.arch, "amd64")].value
```

IMHO we don't need `try` here 🤔 since our default is set to `amd64`
```suggestion
  image_id                             = data.aws_ssm_parameter.al2023_ami[each.value.arch].value
```